presto-spi/src/main/java/com/facebook/presto/spi/function/SqlInvokedFunction.java

```diff
@@ -109,6 +110,12 @@ public String getDescription()
         return description;
     }
 
+    @Override
+    public FunctionImplementationType getFunctionImplementationType()
```

Comment:
This should be a function derived from `RoutineCharacteristics.language`. Actually is it better to expose language rather than implementation type?

Reply:
Language is an attribute for SqlInvokedFunction only. If we expose this information as an agglomerated enum such as in (BUILTIN, SQL, PYTHON, JS...), I would avoid reuse the name `language`, maybe `function type`?

Also, as you mentioned, we may also want to ship SqlInvokedFunctions that are static and packaged within Presto, which type value do we give them? They are both `BUILTIN` and `SQL`, so use a new enum `BUILTIN_SQL`, or `PREPACKAGED_SQL`?

An alternative I would prefer is to introduce 2 columns instead.
- boolean built_in: `true` for hard-coded functions, and pre-packaged `SqlInvokedFunctions`, and `false` for user-defined functions.
- varchar language: `null` for hard-coded functions, and non-null values (SQL, Python, etc...) for all `SqlInvokedFunctions`, including pre-packaged and user-defined.